starknet_py/tests/e2e/deploy/deployer_test.py

```diff
@@ -162,3 +163,48 @@ async def test_create_deployment_call_raw(
 
     assert isinstance(contract_address, int)
     assert contract_address != 0
+
+
+@pytest.mark.asyncio
+async def test_create_deployment_call_raw_supports_seed_0(
+    account,
+    constructor_with_arguments_abi,
+    constructor_with_arguments_class_hash,
+):
+    sample_calldata = {
+        "single_value": 10,
+        "tuple": (1, (2, 3)),
+        "arr": [1, 2, 3],
+        "dict": {"value": 12, "nested_struct": {"value": 99}},
+    }
+    deployer = Deployer()
+
+    raw_calldata = translate_constructor_args(
+        abi=constructor_with_arguments_abi or [], constructor_args=sample_calldata
```

Comment:
Do we need `or []` here?

Reply:
Probably not, also applies to `test_create_deployment_call_raw` which I copy-pasted, can remove

Reply:
> Thanks for the PR! 🚀
> 
> I've ran the checks and it seems that the test passed for `GatewayClient` but failed for `FullNodeClient`.
> 
> Since it might be a deeper issue, I suggest you change the test to use `gateway_client` only for now, so we can merge your bugfix - and we will investigate the `FullNodeClient` issue internally.

changed to `gateway_account`

Reply:
> Probably not, also applies to `test_create_deployment_call_raw` which I copy-pasted, can remove unless as some point `constructor_with_arguments_compiled.json` will not include an ABI for some reason?

Must have been introduced by accident. `constructor_with_arguments_abi` seems to be checking if abi is not None.

Reply:
ok I removed it from both tests